### ISSUE TITLE
Can't Enable Mac Mouse Fix [bug]

### ISSUE THREAD

**Issue by HiramTheHero** [noah-nuebling/Mac-Mouse-Fix#192]
## Description

When I try to click on the square to Enable Mac Mouse Fix, the square doesn't turn blue.

## Steps to reproduce

1. Download Mac Mouse Fix v. 2
2. Do the steps to bypass Mac Security to run the app.
3. Open the App.
4. Try to click the square on the top left.
5. No response.

## Context

Mac Mouse Fix Version: 2.0.0
macOS Version: 12.1
Mouse Model: Logitech

## Additional Comments

Nope. Please help. Lolz. I'm excited to try out the app!

---

### Console Logs

https://pastebin.com/kPhmVRgT

### Crash Reports

–

**Comment by kfr-kevinferreira** [comment]
Hey ! 
I also have this problem :(
I would be too happy to use this app but impossible to activate it.
Similar config on a MacBook pro 14, same app version and OSX version

**Comment by akiftaseen** [comment]
I am not being able to enable the application too no matter how many times I click to enable it. Mac Mouse Fix version 2.2, MacOS version 12.1.

Link to the video of the issue- https://drive.google.com/drive/folders/1oiwofdr2Gs0oVKXIzy9S5PCktLpPrHpu?usp=sharing

**Comment by noah-nuebling** [comment]
Hey guys, thanks for reaching out! And sorry for the late reply.

That video is really helpful. 

Some background: When you check the 'Enable Mac Mouse Fix' checkbox, what actually happens is that Mac Mouse Fix creates a so called 'launchd.plist' file at `~/Library/LaunchAgents/com.nuebling.mac-mouse-fix.helper.plist` which tells macOS to run the 'Mac Mouse Fix Helper' app in the background. This file is automatically loaded by macOS at login. To make macOS load the file and start 'Mac Mouse Fix Helper' immediately, Mac Mouse Fix then also runs the command `launchctl bootstrap gui/$UID ~/Library/LaunchAgents/com.nuebling.mac-mouse-fix.helper.plist`. 

This command fails for some reason. That's what the video tells us. 

If you want to investigate this further, you could run the `launchctl bootstrap gui/$UID ~/Library/LaunchAgents/com.nuebling.mac-mouse-fix.helper.plist` directly in the Terminal to see if that works. It probably won't work. But if you run it as an administrator (`sudo launchctl bootstrap gui/$UID ~/Library/LaunchAgents/com.nuebling.mac-mouse-fix.helper.plist`) it will tell you more about why it's failing.

Let me know if you have any more questions!

**Comment by akiftaseen** [comment]
I was able to get this using your suggestion, but I don't know how to enable this service and couldn't find what unknown error 119 is. Thank you for helping us out.

<img width="917" alt="Screenshot 2022-01-20 at 10 12 29 AM" src="https://user-images.githubusercontent.com/98051538/150272642-7e7ba105-5ced-4c5a-821d-1f0efe285e43.png">

**Comment by noah-nuebling** [comment]
So this looks like the 'Mac Mouse Fix Helper' service has been 'disabled' in the macOS launch services, which makes it impossible to load it.

If you run the command `launchctl enable gui/$UID/mouse.fix.helper` that might fix the issue.

I have no idea how this could have happened. I didn't even know 'disabling' services was a thing. Have you used [LaunchControl](https://www.soma-zone.com/LaunchControl/) or a similar software by any chance?

**Comment by akiftaseen** [comment]
Yes, it worked! Thank you so much! I haven't used LaunchControl, but I use CleanMyMac X, but since this is a very automated and simple software, I cannot tell if it disabled the service. What I can say is that I suspected one of my software was interfering with Mac Mouse Fix and uninstalled and reinstalled by downloading a new Mac Mouse Fix file several times and also restarting my MacBook, but it didn't work and I suspected that maybe MacOS was disabling it after warning that the software couldn't be checked.

Thank you again. I love the software you made.

**Comment by noah-nuebling** [comment]
Awesome glad it works for you!!

Thank you for helping to figure this out, it's probably going to be useful to others as well. And thanks for you input on what might've caused it. 

I might have Mac Mouse Fix 'enable' its service automatically in the future, but I'll have to look into it make sure that's a good idea.

Oh and thanks for the kind words, I'm glad you like the app :)

**Comment by noah-nuebling** [comment]
I just checked, and when you click 'Disable' in CleanMyMac, then this exact issue occurs. So that might really have been the root of the issue.

<img width="514" alt="CleanShot 2022-01-19 at 22 52 26@2x" src="https://user-images.githubusercontent.com/40808343/150281521-b746ff2f-ee38-4dc7-8d71-0f3ddd32c4f3.png">

**Comment by akiftaseen** [comment]
I didn't click any 'disable' button, but since CleanMyMac has the ability to do so, then perhaps it did so while treating the app like malware or sth automatically. It's great that we were able to find the issue! :)

**Comment by lurk26** [comment]
Hi I have this same problem. I ran `launchctl bootstrap gui/$UID ~/Library/LaunchAgents/com.nuebling.mac-mouse-fix.helper.plist`
and got:
`/Users/[username]/Library/LaunchAgents/com.nuebling.mac-mouse-fix.helper.plist: No such file or directory
Bootstrap failed: 2: No such file or directory`

username removed for privacy

**Comment by noah-nuebling** [comment]
Hi @lurk26, that file `~/Library/LaunchAgents/com.nuebling.mac-mouse-fix.helper.plist` should be created by Mac Mouse Fix when you click the 'Enable Mac Mouse Fix' checkbox. (And it will be deleted when you uncheck the checkbox).

There used to be issues where the file couldn't be placed by Mac Mouse Fix when the permissions on the `~/Library/LaunchAgents/` folder were corrupted. But Mac Mouse Fix will now automatically repair those permissions. So I have now idea what might be causing Mac Mouse Fix to fail to create this file. 

If you file a [Bug Report](https://noah-nuebling.github.io/mac-mouse-fix-feedback-assistant/?type=bug-report) and attach some Console Logs that should help bring more light to the situation.

Sorry I don't have much time to work on the project right now but I hope that helps a little!

**Comment by yaoyi2008** [comment]
> launchctl enable gui/$UID/mouse.fix.helper

It works.

**Comment by griffindavidson** [comment]
@noah-nuebling I think I accidentally deleted the LauchDaemon while cleaning up my Library folder and removing uninstalled app's plist files. I've tried reinstalling the app, but the plist isn't reinstalling again. How can I fix this?

**Comment by noah-nuebling** [comment]
Hey @GriffinDavidson, if you’re on the 2.2.1 Beta it doesn’t actually create the launch agent file since there’s a new API in Ventura that doesn’t rely on the file.

If not, then all I can think of right now is that there are some permissions set incorrectly in the library so that MMF can’t create the file.

Hope that helps! Let me know if you have more questions.

**Comment by griffindavidson** [comment]
@noah-nuebling I'm on public release 2.2.0 and using Monterey currently, what permissions are required? I have full disk access and accessibility granted to Mac Mouse Fix, but nothing is working

Update:
After installing the 2.2.1 beta, the app immediately crashes on macOS Monterey 12.5.1 (Intel). [Here's](https://imgur.com/a/1kPmtuU) a screen recording of the crash log if it's any help

Suddenly its now working. I reinstalled the non-beta and tried opening the app before moving to the ~/Applications folder, and the request for accessibility popped up and now its working once again.... weird but I'll take it
